docs/pages/machine-id/access-guides/terraform.mdx

```diff
@@ -173,6 +173,153 @@ Check your Teleport cluster, ensuring the role has been created:
+---
+kind: bot
+version: v2
+metadata:
+  name: github-actions
+spec:
+  token_id: github-actions-bot
+  roles:
+  - terraform
+```
+
+Now we'll combine the available `teleport-actions` and `hashicorp` Actions to run Terraform in CI, using
+tbot to authenticate with the Teleport cluster. The Actions `teleport-actions/setup` and `teleport-actions/auth`
+perform the steps we did in Step 2 above, configuring `tbot` and then starting it up in
+[one-shot mode](../../includes/machine-id/daemon-or-oneshot.mdx).
+
+```yaml
+on:
+ push:
+   branches:
+   - main
+   paths:
+   - terraform/**
+ pull_request:
+   branches:
+   - main
+   paths:
+   - terraform/**
+ 
+env:
+ # verbosity setting for Terraform logs
+ TF_LOG: INFO
+ # Credentials for deployment to AWS
+ AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+ AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+ # S3 bucket for the Terraform state
+ BUCKET_TF_STATE: ${{ secrets.BUCKET_TF_STATE}}
+
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - name: Install Teleport
+        uses: teleport-actions/setup@v1
+        with:
+          version: 15.0.1
+      - name: Authorize against Teleport
+        id: auth
+        uses: teleport-actions/auth@v2
+        with:
+          # Specify the publically accessible address of your Teleport proxy.
+          proxy: tele.example.com:443
+          # Specify the name of the join token for your bot.
+          token: github-actions-bot
+          # Specify the length of time that the generated credentials should be
+          # valid for. This is optional and defaults to "1h"
+          certificate-ttl: 30m
+          # Enable submission of anonymous usage telemetry to Teleport.
+          # See https://goteleport.com/docs/machine-id/reference/telemetry/ for
+          # more information.
+          anonymous-telemetry: 1
+      - name: Checkout the repository to the runner
+        uses: actions/checkout@v3
+  
+      - name: Setup Terraform with specified version on the runner
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.7.3
+      
+      - name: Terraform init
+        id: init
+        run: terraform init -backend-config="bucket=$BUCKET_TF_STATE"
+  
+      - name: Terraform format
+        id: fmt
+        run: terraform fmt -check
+      
+      - name: Terraform validate
+        id: validate
+        run: terraform validate
+  
+      - name: Terraform plan
+        id: plan
+        if: github.event_name == 'pull_request'
+        run: terraform plan -no-color -input=false
+        continue-on-error: true
+      
+      - uses: actions/github-script@v6
+        if: github.event_name == 'pull_request'
+        env:
+          PLAN: "terraform\n${{ steps.plan.outputs.stdout }}"
+        with:
+          script: |
+            const output = `#### Terraform Format and Style 🖌\`${{ steps.fmt.outcome }}\`
```

Comment:
This is nice and fancy but it's also noisy and may distract from what you're actually trying to teach people here (how to use Machine ID with Terraform).

Should we simplify this part?

Reply:
I could take out the Terraform side, and put in a link to Hashi's docs that I pulled them from.

Reply:
Removing this whole addition for now at Paul's suggestion. Resolving.